include/highfive/bits/H5DataType_misc.hpp

```diff
@@ -242,33 +242,59 @@ inline AtomicType<Reference>::AtomicType() {
     _hid = H5Tcopy(H5T_STD_REF_OBJ);
 }
 
+inline size_t find_first_atomic_member_size(hid_t hid)
+{
+    // Recursive exit condition
+    if (H5Tget_class(hid) != H5T_COMPOUND) {
+        return H5Tget_size(hid);
+    }
+
+    auto number_of_members = H5Tget_nmembers(hid);
+    if (number_of_members == -1) {
+        throw DataTypeException("Cannot get members of CompoundType with hid: " +
+                                std::to_string(hid));
+    }
+    if (number_of_members == 0) {
+        throw DataTypeException("No members defined for CompoundType with hid: " +
+                                std::to_string(hid));
+    }
+
+    auto member_type = H5Tget_member_type(hid, 0);
+    return find_first_atomic_member_size(member_type);
```

Comment:
Given the call to H5Tget_member_type one line higher, I think the return line should be replaced with the following:
```
auto size = find_first_atomic_member_size(member_type);
H5Tclose(member_type);
return size;
```

Reply:
I agree, this should be changed.

Reply:
The change has been force-pushed (so that history is a single clean commit. Is that desired, or should it have been a new commit?).